src/python_framework/networkbuilder.py

```diff
@@ -33,9 +32,58 @@ def get_down_connections(
 
     return connections
 
+def get_waterbody_segments(
+    connections = None
+    , terminal_code = 0
+    , waterbody_col = 9
+    , waterbody_null_code = -9999
+    , data_key = r'data'
+    , downstream_key = r'downstream'
+    , upstreams_key = r'upstreams'
+    , verbose = False
+    , debuglevel = 0
+    ):
+
+    if verbose: print('waterbody_set ...')
+    waterbody_set = {con[data_key][waterbody_col] for key, con in connections.items()}
+    waterbody_set.remove(waterbody_null_code)
```

Comment:
this should probably use `set.discard`, unless you want an error thrown if the null code doesn't exist.

Reply:
Thanks -- made that change.